src/gui/constraint_list.rs

```diff
@@ -106,22 +108,96 @@ impl SATApp {
                 .button(RichText::new("Filter").size(text_scale))
                 .clicked()
             {
-                self.state.max_length = apply_max_length(self.state.max_length_input.as_str());
-                if let Some(max_length) = self.state.max_length {
-                    self.filter.by_max_length(max_length);
-                    self.rendered_constraints =
-                        create_tupples_from_constraints(self.filter.get_filtered());
-                }
+                self.state.filter_by_max_length();
+                self.rendered_constraints =
+                    create_tuples_from_constraints(self.state.get_filtered());
             }
             if ui
                 .button(RichText::new("Clear filters").size(text_scale))
                 .clicked()
             {
-                self.filter.clear_all();
+                self.state.clear_filters();
+                self.rendered_constraints =
+                    create_tuples_from_constraints(self.state.get_filtered());
+            }
+        })
+    }
+
+    fn page_length(&mut self, ui: &mut Ui, text_scale: f32) -> egui::InnerResponse<()> {
```

Comment:
This function should be named better, I have no idea what it does based on the name alone. Maybe page_length_input or something?